### ISSUE TITLE
SOP detachedSign() memory usage

### ISSUE THREAD

**Issue by houston3** [vanitasvitae/pgpainless#432]
**Describe the bug**
Thank you for creating this library. It's a pleasure to use. 
My problem is that I'm running out of memory when signing big files (multiple gigabytes) with SOP. Am I doing something wrong here?
```
        SOP sop = new SOPImpl();
        ReadyWithResult<SigningResult> readyWithResult = sop.detachedSign()
                .key(key.getBytes(StandardCharsets.UTF_8))
                .data(new FileInputStream("bigfile.csv"));
```

**Version**

- `pgpainless-core`: 1.6.6
- `pgpainless-sop`: 1.6.6

**To Reproduce**

Use sop.detachedSign() with a big file and watch process memory usage.
```
        SOP sop = new SOPImpl();
        ReadyWithResult<SigningResult> readyWithResult = sop.detachedSign()
                .key(key.getBytes(StandardCharsets.UTF_8))
                .data(new FileInputStream("bigfile.csv"));

```

**Expected behavior**

SOP detachedSign() should be able to handle big files without running out of memory

**Additional context**

As far as I can tell, the problem is that the entire file is buffered into a ByteArrayOutputStream in DetachedSignImpl https://github.com/pgpainless/pgpainless/blob/main/pgpainless-sop/src/main/java/org/pgpainless/sop/DetachedSignImpl.java#L101. 
Would it be possible to allow customization of this output stream? Using e.g. FileOutputStream or OutputStream.nullOutputStream() instead of ByteArrayOutputStream would fix the issue. 
If this sounds like a reasonable feature, I might be able to create a PR.
```
        ReadyWithResult<SigningResult> readyWithResult = sop.detachedSign()
                .key(key.getBytes(StandardCharsets.UTF_8))
                .withOutputStream(myOutputStream) // <- either a new method 
                .data(myData, myOutputStream);    // <- or a new parameter to data()
```

**Comment by vanitasvitae** [comment]
Hey!
Thanks for the bug report!

The buffer was actually not needed at all here.
I fixed the OOM by replacing the buffer with a sink that simply discards the data after passing through the signer stream.

It is not necessary to be able to pass in a custom output stream in this case, since the result of the detached-sign operation are simply the detached signatures, which are sourced from `signingStream.getResult()` and not from the signing streams output (which would be unchanged plaintext data).

Could you build PGPainless from https://github.com/pgpainless/pgpainless/tree/fixSopOOM and see, if the issue is gone with that patch?

It should be enough to clone the repo, check out the `fixSopOOM` branch, call `./gradlew publishToMavenLocal` and then bump your `pgpainless-sop` dependency to `1.6.7-SNAPSHOT`.

**Comment by houston3** [comment]
Thanks for the quick response. That version works perfectly. :+1:

**Comment by vanitasvitae** [comment]
Great, I'll prepare a new release :)

**Comment by vanitasvitae** [comment]
This is fixed with PGPainless 1.6.7

**Comment by houston3** [comment]
Awesome, thank you! :+1:
